dev/src/main/java/org/eclipse/codewind/intellij/ui/toolwindow/UpdateHandler.java

```diff
@@ -0,0 +1,163 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v20.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.codewind.intellij.ui.toolwindow;
+
+import com.intellij.ui.content.Content;
+import org.eclipse.codewind.intellij.core.CodewindApplication;
+import org.eclipse.codewind.intellij.core.IUpdateHandler;
+import org.eclipse.codewind.intellij.core.connection.CodewindConnection;
+import org.eclipse.codewind.intellij.core.connection.ConnectionManager;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Generic Update Handler for any Tool Windows (Overview window, Logs window, Codewind view...)
+ */
+public class UpdateHandler implements IUpdateHandler
+{
+    private static UpdateHandler updateHandler;
+
+    public static UpdateHandler getInstance() {
+        if (updateHandler == null) {
+            updateHandler = new UpdateHandler();
+        }
+        return updateHandler;
+    }
+    private UpdateHandler() {
+        // Empty
+    }
+
+    private static HashMap<AppKey, AppUpdateListener> appListeners = new HashMap<AppKey, AppUpdateListener>();
+
+    @Override
+    public void updateAll() {
+        updateApps();
+    }
+
+    @Override
+    public void updateConnection(CodewindConnection connection) {
+        updateApps(connection);
+    }
+
+    private void updateApps() {
+        ConnectionManager.getManager().activeConnections().stream().forEach(conn -> updateApps(conn));
```

Comment:
This can be replaced with `ConnectionManager.getManager().activeConnections().forEach(this::updateApps);`